src/lib/storage/storage_manager.cpp

```diff
@@ -7,45 +7,57 @@
 
 #include "utils/assert.hpp"
 
+#include <boost/range/combine.hpp>
+
 namespace opossum {
 
 StorageManager& StorageManager::get() {
-  return *(new StorageManager());
-  // A really hacky fix to get the tests to run - replace this with your implementation
+  static StorageManager instance;
+  return instance;
 }
 
 void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> table) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  _table_names.push_back(name);
```

Comment:
Alternatively to using a map, you could also use a pair <std::string, table>, which I would see as an improvement to the current method (but I'd personally still prefer a map as hinted by Daniel).